src/Thruway/Role/Subscriber.php

```diff
@@ -219,13 +251,49 @@ public function subscribe(ClientSession $session, $topicName, callable $callback
             "options"    => $options,
             "deferred"   => $deferred
         ];
-
+        
         array_push($this->subscriptions, $subscription);
-
+        
         $subscribeMsg = new SubscribeMessage($requestId, $options, $topicName);
         $session->sendMessage($subscribeMsg);
 
         return $deferred->promise();
     }
-
+	
+    /**
+     * process unsubscribe
+     * @param ClientSession $session
+     * @param type $topicName
+     * @return boolean
+     */
+    public function unsubscribe(ClientSession $session, $topicName)
+    {
+        $requestId = Utils::getUniqueId();
+        $subscriptionId = false;
+        $deferred  = new Deferred();
+        
+        foreach ($this->subscriptions as $i => $subscription) {
+            if ($subscription["topic_name"] == $topicName) {
+                $subscriptionId = $subscription["subscription_id"];
+                
+                $this->subscriptions[$i]["unsubscribed_request_id"] = $requestId;
+                $this->subscriptions[$i]["unsubscribed_deferred"] = $deferred;
+            }
+        }
```

Comment:
It is possible that more than one subscription to the same uri may exist. Difference routers handle this differently. With the Thruway router, the subscription is going to be a new subscription with a new id - giving two distinct subscriptions with an identical uri. I can't remember exactly what crossbar will do, I believe it will either send a `SUBSCRIBED` with the same subscription id or will send an error back.

This could be further compounded by the fact that subscriptions with the same uri may actually be different because of other reasons (pattern matching comes to mind).

It would be better to handle unsubscribe by using the subscription_id which can be obtained when the promise returned by `subscribe` resolves.